src/spike/spike.service.ts

```diff
@@ -24,15 +12,18 @@ export default class Spike {
      * This function gets a token from spike service
      * @returns the token
      */
-    public async getToken() :Promise<string> {
-        const client = await new spike_proto.Spike(spikeServiceURL, grpc.credentials.createInsecure());
+    public async getToken(): Promise<string> {
+        const client: SpikeClient = new SpikeClient(spikeServiceURL, grpc.credentials.createInsecure());
+        const req: GetSpikeTokenRequest = new GetSpikeTokenRequest();
+        req.setAudience(audiance);
+        req.setGrantType(grantType);
 
         return new Promise((resolve, reject) => {
-            client.GetSpikeToken(spikeReqBody, function (err: Error, response: any) {
+            client.getSpikeToken(req , (err: grpc.ServiceError | null, response: SpikeToken) => {
                 if (err) {
-                    throw new SpikeError(`Error contacting spike: ${err}`);
+                    reject(`Error contacting spike: ${err}`);
```

Comment:
What happens if it fails? Who catches it?
Is it being logged?
Maybe we should try it a few times before pulling the service down.